selftests/functional/test_output.py

```diff
@@ -174,23 +174,25 @@ def _check_output(path, exps, name):
                         i += 1
                         if i == end:
                             break
-                self.assertEqual(i, end, "Failed to find %sth message from\n%s\n"
-                                 "\nin the %s. Either it's missing or in wrong "
-                                 "order.\n%s" % (i, "\n".join(exps), name,
-                                                 output_file_content))
+                exps_text = "\n".join([exp.decode() for exp in exps])
+                error_msg = ("Failed to find message in position %s from\n%s\n"
+                             "\nin the %s. Either it's missing or in wrong "
+                             "order.\n%s" % (i, exps_text, name,
+                                             output_file_content))
```

Comment:
Sure, this version is fine as well. The `i` is there and that's what speedups the search for the first issue.